app/CMakePresets.json

```diff
@@ -0,0 +1,171 @@
+{
+	"version": 3,
+	"cmakeMinimumRequired": {
+	  "major": 3,
+	  "minor": 21,
+	  "patch": 0
+	},
+
+	"configurePresets": [
+	{
+		"name": "common-settings",
+		"description": "List of common settings used by all configurations",
+		"hidden": true,
+		"generator": "Ninja",
+		"environment": {
+			"ZEPHYR_BASE": "${sourceDir}/../../zephyr"
+		}
+	},
+	{
+		"name": "mtl-xtensa-windows",
+		"displayName": "MTL Xtensa",
+		"inherits": [ "mtl-xtensa", "xtensa-toolchain-windows" ],
+		"environment": {
+			"XTENSA_TOOLS_VERSION": "RI-2020.5-win32"
+		}
+	},
+	{
+		"name": "mtl-xtensa-linux",
+		"displayName": "MTL Xtensa",
+		"inherits": [ "mtl-xtensa", "xtensa-toolchain-linux" ],
+		"environment": {
+			"XTENSA_TOOLS_VERSION": "RI-2020.5-linux"
+		}
+	},
+	{
+		"name": "mtl-xtensa",
+		"hidden": true,
+		"description": "Meteorlake build using Xtensa toolchain",
+		"inherits": "common-settings",
+		"binaryDir": "${sourceDir}/../../build-mtl",
+		"cacheVariables": {
+			"BOARD": "intel_adsp_ace15_mtpm",
+			"ZEPHYR_TOOLCHAIN_VARIANT":"xcc"
+		},
+		"environment": {
+			"XTENSA_CORE": "ace10_LX7HiFi4_RI_2020_5"
+		}
+	},
+	{
+		"name": "tgl-xtensa-windows",
+		"displayName": "TGL Xtensa",
+		"inherits": [ "tgl-xtensa", "xtensa-toolchain-windows" ],
+		"environment": {
+			"XTENSA_TOOLS_VERSION": "RG-2017.8-Win32"
+		}
+	},
+	{
+		"name": "tgl-xtensa-linux",
+		"displayName": "TGL Xtensa",
+		"inherits": [ "tgl-xtensa", "xtensa-toolchain-linux" ],
+		"environment": {
+			"XTENSA_TOOLS_VERSION": "RG-2017.8-linux"
+		}
+	},
+	{
+		"name": "tgl-xtensa",
+		"hidden": true,
+		"description": "Tigerlake build using Xtensa toolchain",
+		"inherits": "common-settings",
+		"binaryDir": "${sourceDir}/../../build-tgl",
+		"cacheVariables": {
+			"BOARD": "intel_adsp_cavs25",
+			"ZEPHYR_TOOLCHAIN_VARIANT":"xcc",
+			"OVERLAY_CONFIG": "overlays/tgl/ipc4_overlay.conf"
+		},
+		"environment": {
+			"XTENSA_CORE": "cavs2x_LX6HiFi3_2017_8"
+		}
+	},
+	{
+		"name": "xtensa-toolchain-windows",
+		"hidden": true,
+		"inherits": [ "windows", "xtensa-toolchain" ],
+		"environment": {
+			"XTENSA_INSTALL_PATH": "C:/usr/xtensa"
+		}
+	},
+	{
+		"name": "xtensa-toolchain-linux",
+		"hidden": true,
+		"inherits": [ "linux", "xtensa-toolchain" ],
+		"environment": {
+			"XTENSA_INSTALL_PATH": "~/xtensa"
```

Comment:
Please keep using XTENSA_TOOLS_ROOT for now. One change at a time. Needlessly breaking the build of every CI and every developer is not going to make CMake presets popular.